examples/cpu/fractal.cpp

```diff
@@ -51,7 +51,7 @@ int main(void)
     /* Create an image object which creates the necessary
      * textures and pixel buffer objects to hold the image
      * */
-    fg::Image img(DIMX, DIMY, fg::FG_RGBA, fg::FG_UNSIGNED_BYTE);
+    fg::Image img(DIMX, DIMY, fg::FG_C4, fg::u8);
```

Comment:
On that note, we should think of other formats we might want to support, i think we can the following options:
* Just use base internal formats mentioned in Table 1 of [`glTexImage2D`][ref] - This fits well with current forge API of Image class - no change is required from forge but the user of the forge API needs to handle the format conversion themselves.
* To enable different component layouts(BGR, BGRA) for channels, we have to have another parameter to Image class API so that we can track the distinction between `internalFormat` & `format` parameters of [`glTexImage2D`][ref] API. This new parameter is required because currently we use single variable of type `fg::ColorMode` for both `internalFormat`  & `format` variables of `glTexImage2D`. (It doesn't need to be additional parameter to Image class constructor, but we need some way to identify the difference in layout and format).

[ref]: https://www.opengl.org/sdk/docs/man/html/glTexImage2D.xhtml